### PR TITLE
Feature/fix mark as done

### PR DESCRIPTION
**Pull request by Pellumbengineer**
Before i did some non-sense thing when making the appointment as done. 

So here what i did is i take the previous appointment of current appointment check the actualTime and actualDuration of the previous appointment and then, i set the actual time of the current appointment after the end of the previous appointment. Then calculate the actualDuration and set it.

### CODE REVIEW

**Comment by konpap99** [comment]
Hmm I am not sure if this is right logic but I get what you did! The thing with setting the actualTime as the endOfPrevious appointment time is that you assume that the appointments are consecutive (but that is not always the case)

However your approach works fine I think because using actualTime and actualDuration we can compute when the appointment ended.

One more thing I want you to change so as for the hook to work is the use of updateOne. Instead of using updateOne please use find and then use save. The problem with using updateOne is that I cant check through the hook if the field actualTime was modified. On the other side when using save hook I can. This happens because the updateOne hook give you access to a Query object and the save hook to an IAppointment object

**Comment by Pellumbengineer** [comment]
I assume that the the next appointment will start right after the previous one is done if already passed the time. I will have to check if it is already late if not i will have to set the expected time as actual time right?

**Comment by konpap99** [comment]
I think you are right! You could do  that as well but just pay attention to how you compute the minutes in each case

**Comment by Pellumbengineer** [comment]
i dont actually get how to use save ``` await Appointment.updateOne({ _id: appointment._id }, {
            actualDuration: actualDurationToBeSet,
            actualTime: actualTimeToBeSet,
        });``` 

i need to say update appointment by id and modify these 2 fields. so in save i can not say update this specific appointment. 

I am not sure if save is the right thing to use here. With save i have to create a new appointment.

**Comment by konpap99** [comment]
I think is the suggested way to use save generally, so here is an example:

**Comment by Pellumbengineer** [comment]
I did not know that i could assign actualTime directly like that but it worked. :+1:

File: src/resolvers/mutation.ts [hunk 1]
```diff
@@ -261,10 +277,52 @@ const Mutation: MutationResolvers = {
...
+        const end = new Date();
+        end.setHours(23, 59, 59, 999);
+
+        const previousOfCurrentAppointment = await Appointment.find({
+            'doctorRef.doctorId': appointment.doctorRef.doctorId,
+            expectedTime: { $gte: start, $lt: end },
+            'patientRef.patientId': { $ne: appointment.patientRef.patientId },
+        }).sort({ _id: -1 }).limit(1);
+
+        if(previousOfCurrentAppointment.length < 1 || previousOfCurrentAppointment === undefined) {
```
Comment (konpap99) [inline src/resolvers/mutation.ts:292]: What is the purpose of this if statement?

Comment (Pellumbengineer) [reply src/resolvers/mutation.ts:292]: if there is not any previous appointment at that day (maybe the first appointment of the day)

**Review by konpap99 — APPROVED** [review]

**Pull request merged**